packages/connect-form/src/components/advanced-options-tabs/general/schema-input.tsx

```diff
@@ -0,0 +1,169 @@
+    newConnectionStringUrl.hosts.length > 0
+      ? newConnectionStringUrl.hosts[0].substring(
+          0,
+          newConnectionStringUrl.hosts[0].indexOf(':') === -1
+            ? undefined
+            : newConnectionStringUrl.hosts[0].indexOf(':')
+        )
+      : defaultHostname;
+  newConnectionStringUrl.hosts = [newHost];
+
+  const newConnectionString = newConnectionStringUrl.toString();
+  newConnectionStringUrl = new ConnectionStringUrl(
+    newConnectionString.replace('mongodb://', 'mongodb+srv://')
+  );
+
+  // SRV connections can't have directConnection set.
+  if (newConnectionStringUrl.searchParams.get('directConnection')) {
+    newConnectionStringUrl.searchParams.delete('directConnection');
+  }
+
+  return newConnectionStringUrl;
+}
+
+function updateConnectionStringSchema(
+  connectionStringUrl: ConnectionStringUrl,
+  schema: MONGODB_SCHEMA
+): ConnectionStringUrl {
+  if (schema === MONGODB_SCHEMA.MONGODB) {
+    return updateConnectionStringToStandard(connectionStringUrl);
+  } else {
+    return updateConnectionStringToSRV(connectionStringUrl);
+  }
+}
+
+function SchemaInput({
+  connectionStringUrl,
+  schemaConversionError,
+  setConnectionField,
+  setConnectionStringUrl,
+}: {
+  connectionStringUrl: ConnectionStringUrl;
+  schemaConversionError: string | null;
+  setConnectionField: SetConnectionField;
+  setConnectionStringUrl: (connectionStringUrl: ConnectionStringUrl) => void;
+}): React.ReactElement {
+  const { isSRV } = connectionStringUrl;
+
+  function onChangeConnectionSchema(
+    event: React.ChangeEvent<HTMLInputElement>
+  ) {
+    try {
+      const newConnectionStringUrl = updateConnectionStringSchema(
+        connectionStringUrl,
+        event.target.value as MONGODB_SCHEMA
+      );
+
+      setConnectionStringUrl(newConnectionStringUrl);
+    } catch (err) {
+      setConnectionField({
+        type: 'set-connection-string-field',
+        fieldName: 'isSRV',
+        value: {
+          conversionError: `Error updating connection schema: ${
+            (err as Error).message
+          }`,
+        },
+      });
+    }
+  }
+
+  function onCloseSchemaConversionError() {
+    setConnectionField({
+      type: 'set-connection-string-field',
+      fieldName: 'isSRV',
+      value: {
+        conversionError: null,
+      },
+    });
+  }
+
+  return (
+    <>
+      <Label htmlFor="connection-schema-radio-box-group">Schema</Label>
+      <RadioBoxGroup
+        id="connection-schema-radio-box-group"
+        value={isSRV ? MONGODB_SCHEMA.MONGODB_SRV : MONGODB_SCHEMA.MONGODB}
+        onChange={onChangeConnectionSchema}
+      >
+        <RadioBox value={MONGODB_SCHEMA.MONGODB}>mongodb</RadioBox>
+        <RadioBox value={MONGODB_SCHEMA.MONGODB_SRV}>mongodb+srv</RadioBox>
+      </RadioBoxGroup>
+      <Description className={descriptionStyles}>
+        {isSRV ? srvSchemaDescription : regularSchemaDescription}
+      </Description>
+      {schemaConversionError && (
+        <Banner
```

Comment:
Honestly I'm not sure, I added the error handling a bit early on when a user could put in an invalid blank hostname and it caused an error here. I think it shouldn't/can't happen. That being said I think it's nice for a user for this error to be caught if they get into a situation we haven't though of. Think we should remove it? https://jira.mongodb.org/browse/COMPASS-5075 will help with handling unexpected errors, but it's nice if we can catch them even closer.

Reply:
I see, yeah I also cannot find an error that should happen there, all the errors for the schema would break the connection string, and then the form would be disabled. 

We mapped what should be a general error and what should be inline here, feel free to update this table: https://docs.google.com/spreadsheets/d/14jRnz50BzqxS7MSPXyCWjU_CqMm3U6M2AhJlpvgPFDk